{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import re\n",
    "import scipy.stats as stats\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "with open(\"recalibrated_variants.postCGP.Gfiltered.Optom_subset.vcf\", 'rt') as first_file:\n",
    "    optom = {}\n",
    "    info = []\n",
    "    # create a list of locations in optom\n",
    "    snp = {}\n",
    "    for line in first_file:\n",
    "        if line.startswith('#'):\n",
    "            continue\n",
    "        row = line.split('\\t')\n",
    "        ### remove the chr from the chromosomes\n",
    "        tmp = [stuff.replace(\"chr\", \"\") for stuff in row]\n",
    "        ### create an index which is chr:bp\n",
    "        chr_bp = tmp[0:2]\n",
    "        index = ':'.join(map(str, chr_bp))\n",
    "        ### output index, rsnumber, a1, a2, info\n",
    "        info = [ tmp[2], tmp[3], tmp[4], tmp[6]]\n",
    "        info.extend(tmp[7].split(';')[0:3])\n",
    "        #if tmp[7].split(';')[3] != \"set=FilteredInAll\" :\n",
    "        #    if tmp[9].split(':')[3] != \"lowGQ\" : \n",
    "        if index not in optom:\n",
    "            optom[index] = info\n",
    "        if tmp[2] not in snp and tmp[2] != '.':\n",
    "            snp[tmp[2]] = 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6084320"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(optom)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rs5862681', 'A', 'AT', 'PASS', 'AC=30', 'AF=0.375', 'AN=30']"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "optom['4:144938033']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr:bp</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>optom_flag</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:100000625</td>\n",
       "      <td>rs7899632</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=0</td>\n",
       "      <td>AF=0.025</td>\n",
       "      <td>AN=30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:100000645</td>\n",
       "      <td>rs61875309</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=0</td>\n",
       "      <td>AF=0.013</td>\n",
       "      <td>AN=30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10:100001277</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=0</td>\n",
       "      <td>AF=0.013</td>\n",
       "      <td>AN=30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10:100001507</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=0</td>\n",
       "      <td>AF=0.013</td>\n",
       "      <td>AN=30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10:100002399</td>\n",
       "      <td>rs8181398</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=30</td>\n",
       "      <td>AF=1.00</td>\n",
       "      <td>AN=30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr:bp       rs_id A1 A2 optom_flag     AC        AF     AN\n",
       "0  10:100000625   rs7899632  A  G       PASS   AC=0  AF=0.025  AN=30\n",
       "1  10:100000645  rs61875309  A  C       PASS   AC=0  AF=0.013  AN=30\n",
       "2  10:100001277           .  G  T       PASS   AC=0  AF=0.013  AN=30\n",
       "3  10:100001507           .  A  C       PASS   AC=0  AF=0.013  AN=30\n",
       "4  10:100002399   rs8181398  A  G       PASS  AC=30   AF=1.00  AN=30"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "optom_all = pd.DataFrame(optom).T.reset_index()\n",
    "optom_all.rename(columns={'index': 'chr:bp', 0: 'rs_id', 1:'A1', 2:'A2', 3:\"optom_flag\", 4:\"AC\", 5:\"AF\", 6:\"AN\"}, inplace=True)\n",
    "optom_all.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "optom_all.to_csv('optom_data_vsqr.csv', index=False, header=True, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get optom genos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "with open(\"recalibrated_variants.postCGP.Gfiltered.Optom_subset.vcf\", 'rt') as f:\n",
    "    optom_geno = {}\n",
    "    info = []\n",
    "    # create a list of locations in optom\n",
    "    for line in f:\n",
    "        if line.startswith('#'):\n",
    "            continue\n",
    "        row = line.split('\\t')\n",
    "        tmp = [stuff.replace(\"chr\", \"\") for stuff in row]\n",
    "        ### create an index which is chr:bp\n",
    "        chr_bp = tmp[0:2]\n",
    "        index = ':'.join(map(str, chr_bp))\n",
    "        if index not in optom_geno:\n",
    "            homo_major = 0\n",
    "            hets = 0\n",
    "            homo_minor = 0\n",
    "            lowGQ_count = 0\n",
    "            for x in range(9,24) :\n",
    "                data = tmp[x].split(':')\n",
    "                #print(data)\n",
    "                if data[0] == \"0/0\" :\n",
    "                    homo_major += 1\n",
    "                elif data[0] == \"0/1\" :\n",
    "                    hets += 1\n",
    "                elif data[0] == \"1/1\":\n",
    "                    homo_minor += 1\n",
    "                if len(data) > 3:\n",
    "                    if data[3] == 'lowGQ':\n",
    "                        lowGQ_count += 1\n",
    "            info = [tmp[2], tmp[3], tmp[4], homo_major, hets, homo_minor, lowGQ_count]\n",
    "            optom_geno[index] = info "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rs5862681', 'A', 'AT', 0, 0, 15, 4]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "optom_geno['4:144938033']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr:bp</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>homo_major</th>\n",
       "      <th>hets</th>\n",
       "      <th>homo_minor</th>\n",
       "      <th>lowGQ_numbers</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:100000625</td>\n",
       "      <td>rs7899632</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:100000645</td>\n",
       "      <td>rs61875309</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10:100001277</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10:100001507</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10:100002399</td>\n",
       "      <td>rs8181398</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr:bp       rs_id A1 A2 homo_major hets homo_minor lowGQ_numbers\n",
       "0  10:100000625   rs7899632  A  G         15    0          0            15\n",
       "1  10:100000645  rs61875309  A  C         15    0          0            15\n",
       "2  10:100001277           .  G  T         15    0          0            15\n",
       "3  10:100001507           .  A  C         15    0          0            15\n",
       "4  10:100002399   rs8181398  A  G          0    0         15             0"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genos = pd.DataFrame(optom_geno).T.reset_index()\n",
    "genos.rename(columns={'index': 'chr:bp', 0: 'rs_id', 1:'A1', 2:'A2', 3:'homo_major', 4:'hets', \n",
    "                      5:'homo_minor', 6:'lowGQ_numbers' }, inplace=True)\n",
    "genos.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "genos.to_csv('optom_genotpyes_vsqr.csv', index=False, header=True, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Association with ALL populations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Allele counts to keep\n",
    "list_names =[\"AC\", \"AN\", \"AC_NFE\", \"AN_NFE\", \"AF\"]\n",
    "pattern = re.compile(\"|\".join(list_names))\n",
    "\n",
    "#not_pass_qc_gnomad = {}\n",
    "\n",
    "with open(\"gnomad.exomes.r2.0.2.sites.vcf\", 'rt') as EXAC:\n",
    "    gnomad = {}\n",
    "    for line in EXAC:\n",
    "        if line.startswith('#'):\n",
    "            continue\n",
    "        row = line.split('\\t')\n",
    "        ### create an index which is chr:bp\n",
    "        chr_bp = row[0:2]\n",
    "        index = ':'.join(map(str, chr_bp))\n",
    "        ### keep only those that are in optom:\n",
    "        if index in optom:\n",
    "            #if row[6] != \"PASS\":\n",
    "            #    if index not in not_pass_qc_gnomad:\n",
    "            #        not_pass_qc_gnomad[index] = [ row[2], row[6] ]\n",
    "            #    continue\n",
    "            ### output index, rsnumber, a1, a2, info\n",
    "            output = [ row[2], row[3], row[4], row[6], row[-1].split(';')[-1] ]\n",
    "            info = row[7].split(';')\n",
    "            keep_info = [item for item in info if re.match(pattern, item)]\n",
    "            output.extend(keep_info)\n",
    "            #print(info)\n",
    "            if index not in gnomad:\n",
    "                gnomad[index] = output\n",
    "        elif row[2] in snp:\n",
    "            if index not in gnomad:\n",
    "                ### output index, rsnumber, a1, a2, info\n",
    "                output = [ index, row[2], row[3], row[4], row[6], row[-1].split(';')[-1] ]\n",
    "                info = row[7].split(';')\n",
    "                keep_info = [item for item in info if re.match(pattern, item)]\n",
    "                output.extend(keep_info)\n",
    "                gnomad[index] = output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "188436"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(gnomad)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rs4041247',\n",
       " 'AAGAGAG',\n",
       " 'AAGAGAGAG,AAGAGAGAGAG,AAGAG,AAG,AAGAGAGAGAGAG,A',\n",
       " 'PASS',\n",
       " 'lcr\\n',\n",
       " 'AC=1823,921,4177,758,133,1801',\n",
       " 'AF=6.56180e-02,3.31510e-02,1.50349e-01,2.72839e-02,4.78727e-03,6.48261e-02',\n",
       " 'AN=27782',\n",
       " 'AC_AFR=78,90,320,61,32,43',\n",
       " 'AC_AMR=112,77,394,74,11,83',\n",
       " 'AC_ASJ=25,20,69,16,0,18',\n",
       " 'AC_EAS=98,189,343,106,42,18',\n",
       " 'AC_FIN=342,124,703,117,5,427',\n",
       " 'AC_NFE=1021,346,1983,315,31,1033',\n",
       " 'AC_OTH=32,20,79,12,2,28',\n",
       " 'AC_SAS=115,55,286,57,10,151',\n",
       " 'AC_Male=1007,490,2308,429,73,1019',\n",
       " 'AC_Female=816,431,1869,329,60,782',\n",
       " 'AN_AFR=2136',\n",
       " 'AN_AMR=2198',\n",
       " 'AN_ASJ=452',\n",
       " 'AN_EAS=1246',\n",
       " 'AN_FIN=4478',\n",
       " 'AN_NFE=15044',\n",
       " 'AN_OTH=526',\n",
       " 'AN_SAS=1702',\n",
       " 'AN_Male=15080',\n",
       " 'AN_Female=12702',\n",
       " 'AF_AFR=3.65169e-02,4.21348e-02,1.49813e-01,2.85581e-02,1.49813e-02,2.01311e-02',\n",
       " 'AF_AMR=5.09554e-02,3.50318e-02,1.79254e-01,3.36670e-02,5.00455e-03,3.77616e-02',\n",
       " 'AF_ASJ=5.53097e-02,4.42478e-02,1.52655e-01,3.53982e-02,0.00000e+00,3.98230e-02',\n",
       " 'AF_EAS=7.86517e-02,1.51685e-01,2.75281e-01,8.50722e-02,3.37079e-02,1.44462e-02',\n",
       " 'AF_FIN=7.63734e-02,2.76909e-02,1.56990e-01,2.61277e-02,1.11657e-03,9.53551e-02',\n",
       " 'AF_NFE=6.78676e-02,2.29992e-02,1.31813e-01,2.09386e-02,2.06062e-03,6.86652e-02',\n",
       " 'AF_OTH=6.08365e-02,3.80228e-02,1.50190e-01,2.28137e-02,3.80228e-03,5.32319e-02',\n",
       " 'AF_SAS=6.75676e-02,3.23149e-02,1.68038e-01,3.34900e-02,5.87544e-03,8.87192e-02',\n",
       " 'AF_Male=6.67772e-02,3.24934e-02,1.53050e-01,2.84483e-02,4.84085e-03,6.75729e-02',\n",
       " 'AF_Female=6.42419e-02,3.39317e-02,1.47142e-01,2.59014e-02,4.72367e-03,6.15651e-02',\n",
       " 'AC_raw=17473,10203,26538,5375,1614,7740',\n",
       " 'AN_raw=223552',\n",
       " 'AF_raw=7.81608e-02,4.56404e-02,1.18711e-01,2.40436e-02,7.21980e-03,3.46228e-02',\n",
       " 'AC_POPMAX=98,189,343,106,42,427',\n",
       " 'AN_POPMAX=1246,1246,1246,1246,1246,4478',\n",
       " 'AF_POPMAX=7.86517e-02,1.51685e-01,2.75281e-01,8.50722e-02,3.37079e-02,9.53551e-02']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnomad['5:172745269']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'6.56180e-02'"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "re.split(\"=|,\", gnomad['5:172745269'][6])[1]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "1) Match keys between the dictionaries\n",
    "2) ensure the A1 and A1 are the same\n",
    "    - gnomad has multiple variants per list -- how best to separate these?\n",
    "3) Create Fisher's exact table for testing\n",
    "    - case AC, case AN-AC, control AC, control AN-AC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "gnomad_Set = set(gnomad)\n",
    "optom_Set = set(optom)\n",
    "\n",
    "results = {}\n",
    "results_gws = {}\n",
    "bad_snps = {}\n",
    "for name in optom_Set.intersection(gnomad_Set):\n",
    "    if optom[name][1] == gnomad[name][1] and optom[name][2] == gnomad[name][2]:\n",
    "        if (re.split(\"=|,\", gnomad[name][6])[1] != \".\") and (float(re.split(\"=|,\", gnomad[name][6])[1]) <= 0.05):\n",
    "            #print(optom[name][1], gnomad[name][1], optom[name][2], gnomad[name][2])\n",
    "            case_AC = int(re.split(\"=|,\", optom[name][4])[1])\n",
    "            case_AN = int(optom[name][6].split('=')[1])\n",
    "            control_AC = int(re.split(\"=|,\", gnomad[name][5])[1])\n",
    "            control_AN = int(gnomad[name][7].split('=')[1])\n",
    "            #print(case_AC, case_AN, control_AC, control_AN)\n",
    "            oddsratio, pvalue = stats.fisher_exact([[case_AC, case_AN-case_AC], [control_AC, control_AN-control_AC]], alternative='two-sided')\n",
    "            info = [case_AC, case_AN, control_AC, control_AN, gnomad[name][3], re.split(\"=|,\", gnomad[name][6])[1], oddsratio, pvalue, gnomad[name][4]]\n",
    "            if name not in results:\n",
    "                results[name] = info\n",
    "            if pvalue < 0.000005 :\n",
    "                if name not in results_gws:\n",
    "                    results_gws[name] = info\n",
    "    else:\n",
    "        data = [optom[name][1],  optom[name][2], gnomad[name][1], gnomad[name][2]]\n",
    "        bad_snps[name] = data\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "53939"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(bad_snps)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "bad_snps_still = {}\n",
    "\n",
    "for name in bad_snps:\n",
    "    if optom[name][1] == gnomad[name][1] and optom[name][2] == gnomad[name][2].split(',')[0] :\n",
    "        if (re.split(\"=|,\", gnomad[name][6])[1] != \".\") and (float(re.split(\"=|,\", gnomad[name][6])[1]) <= 0.05):\n",
    "            #print(optom[name][1], gnomad[name][1], optom[name][2], gnomad[name][2])\n",
    "            case_AC = int(re.split(\"=|,\", optom[name][4])[1])\n",
    "            case_AN = int(optom[name][6].split('=')[1])\n",
    "            control_AC = int(re.split(\"=|,\", gnomad[name][5])[1])\n",
    "            control_AN = int(gnomad[name][7].split('=')[1])\n",
    "            #print(case_AC, case_AN, control_AC, control_AN)\n",
    "            oddsratio, pvalue = stats.fisher_exact([[case_AC, case_AN-case_AC], [control_AC, control_AN-control_AC]], alternative='two-sided')\n",
    "            info = [case_AC, case_AN, control_AC, control_AN, gnomad[name][3], re.split(\"=|,\", gnomad[name][6])[1], oddsratio, pvalue, gnomad[name][4]]\n",
    "            if name not in results:\n",
    "                results[name] = info\n",
    "            if pvalue < 0.000005 :\n",
    "                if name not in results_gws:\n",
    "                    results_gws[name] = info\n",
    "    elif len(re.split(\"=|,\", optom[name][3])) > 2:\n",
    "        if optom[name][1] == gnomad[name][1] and optom[name][2].split(',')[0] == gnomad[name][2].split(',')[0] :\n",
    "            if (re.split(\"=|,\", gnomad[name][6])[1] != \".\") and (float(re.split(\"=|,\", gnomad[name][6])[1]) <= 0.05):\n",
    "                #print(optom[name][1], gnomad[name][1], optom[name][2], gnomad[name][2])\n",
    "                case_AC = int(re.split(\"=|,\", optom[name][4])[1])\n",
    "                case_AN = int(optom[name][6].split('=')[1])\n",
    "                control_AC = int(re.split(\"=|,\", gnomad[name][5])[1])\n",
    "                control_AN = int(gnomad[name][7].split('=')[1])\n",
    "                #print(case_AC, case_AN, control_AC, control_AN)\n",
    "                oddsratio, pvalue = stats.fisher_exact([[case_AC, case_AN-case_AC], [control_AC, control_AN-control_AC]], alternative='two-sided')\n",
    "                info = [case_AC, case_AN, control_AC, control_AN, gnomad[name][3], re.split(\"=|,\", gnomad[name][6])[1], oddsratio, pvalue, gnomad[name][4]]\n",
    "                new_name  = name + \"_1\"\n",
    "                if new_name not in results:\n",
    "                    results[new_name] = info\n",
    "                if pvalue < 0.000005 :\n",
    "                    if new_name not in results_gws:\n",
    "                        results_gws[new_name] = info\n",
    "    elif len(re.split(\"=|,\", optom[name][3])) > 3:\n",
    "        if optom[name][1] == gnomad[name][1] and optom[name][2].split(',')[1] == gnomad[name][2].split(',')[1] :\n",
    "            if (re.split(\"=|,\", gnomad[name][6])[1] != \".\") and (float(re.split(\"=|,\", gnomad[name][6])[2]) <= 0.05):\n",
    "                #print(optom[name][1], gnomad[name][1], optom[name][2], gnomad[name][2])\n",
    "                case_AC = int(re.split(\"=|,\", optom[name][4])[2])\n",
    "                case_AN = int(optom[name][6].split('=')[1])\n",
    "                control_AC = int(re.split(\"=|,\", gnomad[name][5])[2])\n",
    "                control_AN = int(gnomad[name][7].split('=')[1])\n",
    "                #print(case_AC, case_AN, control_AC, control_AN)\n",
    "                oddsratio, pvalue = stats.fisher_exact([[case_AC, case_AN-case_AC], [control_AC, control_AN-control_AC]], alternative='two-sided')\n",
    "                info = [case_AC, case_AN, control_AC, control_AN, gnomad[name][3], re.split(\"=|,\", gnomad[name][6])[2], oddsratio, pvalue, gnomad[name][4]]\n",
    "                new_name  = name + \"_2\"\n",
    "                if new_name not in results:\n",
    "                    results[new_name] = info\n",
    "                if pvalue < 0.000005 :\n",
    "                    if new_name not in results_gws:\n",
    "                        results_gws[new_name] = info   \n",
    "    elif len(re.split(\"=|,\", optom[name][3])) > 4:\n",
    "        if optom[name][1] == gnomad[name][1] and optom[name][2].split(',')[2] == gnomad[name][2].split(',')[2] :\n",
    "            if (re.split(\"=|,\", gnomad[name][6])[1] != \".\") and (float(re.split(\"=|,\", gnomad[name][6])[3]) <= 0.05):\n",
    "                #print(optom[name][1], gnomad[name][1], optom[name][2], gnomad[name][2])\n",
    "                case_AC = int(re.split(\"=|,\", optom[name][4])[3])\n",
    "                case_AN = int(optom[name][6].split('=')[1])\n",
    "                control_AC = int(re.split(\"=|,\", gnomad[name][5])[3])\n",
    "                control_AN = int(gnomad[name][7].split('=')[1])\n",
    "                #print(case_AC, case_AN, control_AC, control_AN)\n",
    "                oddsratio, pvalue = stats.fisher_exact([[case_AC, case_AN-case_AC], [control_AC, control_AN-control_AC]], alternative='two-sided')\n",
    "                info = [case_AC, case_AN, control_AC, control_AN, gnomad[name][3], re.split(\"=|,\", gnomad[name][6])[3], oddsratio, pvalue, gnomad[name][4]]\n",
    "                new_name  = name + \"_3\"\n",
    "                if new_name not in results:\n",
    "                    results[new_name] = info\n",
    "                if pvalue < 0.000005 :\n",
    "                    if new_name not in results_gws:\n",
    "                        results_gws[new_name] = info   \n",
    "    else: \n",
    "        data = [optom[name][1],  optom[name][2], gnomad[name][1], gnomad[name][2]]\n",
    "        bad_snps_still[name] = data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "14070"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(bad_snps_still)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "79832"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "all_keys = set(list(results.keys()) + list(optom.keys()))\n",
    "\n",
    "optom_sig = {}\n",
    "for key in all_keys:\n",
    "    if key in results and key in optom:\n",
    "        optom_sig[key] = optom[key] + results[key]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr:bp</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>flag_optom</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>filter</th>\n",
       "      <th>case_AC</th>\n",
       "      <th>case_AN</th>\n",
       "      <th>control_AC</th>\n",
       "      <th>control_AN</th>\n",
       "      <th>flag_gnomad</th>\n",
       "      <th>gnomad_AF</th>\n",
       "      <th>oddsratio</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>annotations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:100011298</td>\n",
       "      <td>rs138062420</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=0</td>\n",
       "      <td>AF=0.013</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-2.492e+00</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>602</td>\n",
       "      <td>243964</td>\n",
       "      <td>PASS</td>\n",
       "      <td>2.46758e-03</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>CSQ=A|intron_variant|MODIFIER|LOXL4|ENSG000001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:100144676</td>\n",
       "      <td>rs191156429</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=0</td>\n",
       "      <td>AF=0.013</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-6.391e+00</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>66</td>\n",
       "      <td>225918</td>\n",
       "      <td>PASS</td>\n",
       "      <td>2.92141e-04</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>CSQ=T|intron_variant|MODIFIER|PYROXD2|ENSG0000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10:100148117</td>\n",
       "      <td>rs147555747</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1</td>\n",
       "      <td>AF=0.013</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-2.965e+00</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>435</td>\n",
       "      <td>245902</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1.76900e-03</td>\n",
       "      <td>19.4583</td>\n",
       "      <td>0.0518436</td>\n",
       "      <td>CSQ=T|missense_variant|MODERATE|PYROXD2|ENSG00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10:100150830</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=0</td>\n",
       "      <td>AF=0.013</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-2.322e+00</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>246</td>\n",
       "      <td>243128</td>\n",
       "      <td>RF</td>\n",
       "      <td>1.01181e-03</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>CSQ=A|missense_variant|MODERATE|PYROXD2|ENSG00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10:100167323</td>\n",
       "      <td>rs79218397</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1</td>\n",
       "      <td>AF=0.013</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=1.96</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>1012</td>\n",
       "      <td>243762</td>\n",
       "      <td>PASS</td>\n",
       "      <td>4.15159e-03</td>\n",
       "      <td>8.27143</td>\n",
       "      <td>0.117435</td>\n",
       "      <td>CSQ=A|intron_variant|MODIFIER|PYROXD2|ENSG0000...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr:bp        rs_id A1 A2 flag_optom    AC        AF     AN  \\\n",
       "0  10:100011298  rs138062420  G  A       PASS  AC=0  AF=0.013  AN=30   \n",
       "1  10:100144676  rs191156429  C  T       PASS  AC=0  AF=0.013  AN=30   \n",
       "2  10:100148117  rs147555747  C  T       PASS  AC=1  AF=0.013  AN=30   \n",
       "3  10:100150830            .  G  A       PASS  AC=0  AF=0.013  AN=30   \n",
       "4  10:100167323   rs79218397  G  A       PASS  AC=1  AF=0.013  AN=30   \n",
       "\n",
       "                    filter case_AC case_AN control_AC control_AN flag_gnomad  \\\n",
       "0  BaseQRankSum=-2.492e+00       0      30        602     243964        PASS   \n",
       "1  BaseQRankSum=-6.391e+00       0      30         66     225918        PASS   \n",
       "2  BaseQRankSum=-2.965e+00       1      30        435     245902        PASS   \n",
       "3  BaseQRankSum=-2.322e+00       0      30        246     243128          RF   \n",
       "4        BaseQRankSum=1.96       1      30       1012     243762        PASS   \n",
       "\n",
       "     gnomad_AF oddsratio     pvalue  \\\n",
       "0  2.46758e-03         0          1   \n",
       "1  2.92141e-04         0          1   \n",
       "2  1.76900e-03   19.4583  0.0518436   \n",
       "3  1.01181e-03         0          1   \n",
       "4  4.15159e-03   8.27143   0.117435   \n",
       "\n",
       "                                         annotations  \n",
       "0  CSQ=A|intron_variant|MODIFIER|LOXL4|ENSG000001...  \n",
       "1  CSQ=T|intron_variant|MODIFIER|PYROXD2|ENSG0000...  \n",
       "2  CSQ=T|missense_variant|MODERATE|PYROXD2|ENSG00...  \n",
       "3  CSQ=A|missense_variant|MODERATE|PYROXD2|ENSG00...  \n",
       "4  CSQ=A|intron_variant|MODIFIER|PYROXD2|ENSG0000...  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_sig = pd.DataFrame(optom_sig).T.reset_index()\n",
    "test_sig.rename(columns={'index': 'chr:bp', 0: 'rs_id', 1:'A1', 2:'A2', 3:'flag_optom', 4:'AC', 5:'AF', 6:'AN', \n",
    "                   7:'filter',  8:'case_AC', 9:'case_AN', 10:'control_AC', 11:'control_AN', \n",
    "                   12:'flag_gnomad', 13:'gnomad_AF', 14:'oddsratio', 15:'pvalue', 16:'annotations'}, inplace=True)\n",
    "test_sig.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "79832"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(test_sig)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "test_sig.to_csv('optom_results_exac.csv', index=False, header=True, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "all_keys = set(list(results.keys()) + list(optom.keys()))\n",
    "\n",
    "optom_sig_gws = {}\n",
    "for key in all_keys:\n",
    "    if key in results_gws and key in optom:\n",
    "        optom_sig_gws[key] = optom[key] + results_gws[key]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2545"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(optom_sig_gws)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr:bp</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>flag_optom</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>filter</th>\n",
       "      <th>case_AC</th>\n",
       "      <th>case_AN</th>\n",
       "      <th>control_AC</th>\n",
       "      <th>control_AN</th>\n",
       "      <th>flag_gnomad</th>\n",
       "      <th>gnomad_AF</th>\n",
       "      <th>oddsratio</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>annotations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:116698112</td>\n",
       "      <td>rs199739741</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=9</td>\n",
       "      <td>AF=0.113</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-3.965e+00</td>\n",
       "      <td>9</td>\n",
       "      <td>30</td>\n",
       "      <td>3424</td>\n",
       "      <td>162722</td>\n",
       "      <td>RF</td>\n",
       "      <td>2.10420e-02</td>\n",
       "      <td>19.9388</td>\n",
       "      <td>7.84166e-09</td>\n",
       "      <td>CSQ=C|missense_variant|MODERATE|TRUB1|ENSG0000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:118927198</td>\n",
       "      <td>.</td>\n",
       "      <td>TGGTCCCGGCGCCCGGCCTGTGTGGTGCTCAGATCAGACTTATCCA...</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=8</td>\n",
       "      <td>AF=0.100</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-1.360e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>30</td>\n",
       "      <td>89</td>\n",
       "      <td>6468</td>\n",
       "      <td>RF</td>\n",
       "      <td>1.37600e-02</td>\n",
       "      <td>26.0633</td>\n",
       "      <td>8.22806e-09</td>\n",
       "      <td>CSQ=-|intron_variant&amp;non_coding_transcript_var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10:135438959</td>\n",
       "      <td>.</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>VQSRTrancheINDEL99.90to100.00</td>\n",
       "      <td>AC=6</td>\n",
       "      <td>AF=0.225</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-3.050e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>30</td>\n",
       "      <td>1</td>\n",
       "      <td>244284</td>\n",
       "      <td>PASS</td>\n",
       "      <td>4.09360e-06</td>\n",
       "      <td>61070.8</td>\n",
       "      <td>1.40719e-23</td>\n",
       "      <td>segdup\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10:135439015</td>\n",
       "      <td>rs75470891</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=4</td>\n",
       "      <td>AF=0.075</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-6.700e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>30</td>\n",
       "      <td>729</td>\n",
       "      <td>238948</td>\n",
       "      <td>RF</td>\n",
       "      <td>3.05087e-03</td>\n",
       "      <td>50.2731</td>\n",
       "      <td>2.25792e-06</td>\n",
       "      <td>segdup\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10:135439049</td>\n",
       "      <td>rs200347477</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=5</td>\n",
       "      <td>AF=0.113</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=1.89</td>\n",
       "      <td>5</td>\n",
       "      <td>30</td>\n",
       "      <td>360</td>\n",
       "      <td>227178</td>\n",
       "      <td>RF</td>\n",
       "      <td>1.58466e-03</td>\n",
       "      <td>126.01</td>\n",
       "      <td>1.4352e-09</td>\n",
       "      <td>segdup\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr:bp        rs_id  \\\n",
       "0  10:116698112  rs199739741   \n",
       "1  10:118927198            .   \n",
       "2  10:135438959            .   \n",
       "3  10:135439015   rs75470891   \n",
       "4  10:135439049  rs200347477   \n",
       "\n",
       "                                                  A1 A2  \\\n",
       "0                                                  A  C   \n",
       "1  TGGTCCCGGCGCCCGGCCTGTGTGGTGCTCAGATCAGACTTATCCA...  T   \n",
       "2                                                 AC  A   \n",
       "3                                                  T  C   \n",
       "4                                                  G  C   \n",
       "\n",
       "                      flag_optom    AC        AF     AN  \\\n",
       "0    VQSRTrancheSNP99.90to100.00  AC=9  AF=0.113  AN=30   \n",
       "1                           PASS  AC=8  AF=0.100  AN=30   \n",
       "2  VQSRTrancheINDEL99.90to100.00  AC=6  AF=0.225  AN=30   \n",
       "3    VQSRTrancheSNP99.90to100.00  AC=4  AF=0.075  AN=30   \n",
       "4    VQSRTrancheSNP99.90to100.00  AC=5  AF=0.113  AN=30   \n",
       "\n",
       "                    filter case_AC case_AN control_AC control_AN flag_gnomad  \\\n",
       "0  BaseQRankSum=-3.965e+00       9      30       3424     162722          RF   \n",
       "1  BaseQRankSum=-1.360e-01       8      30         89       6468          RF   \n",
       "2  BaseQRankSum=-3.050e-01       6      30          1     244284        PASS   \n",
       "3  BaseQRankSum=-6.700e-02       4      30        729     238948          RF   \n",
       "4        BaseQRankSum=1.89       5      30        360     227178          RF   \n",
       "\n",
       "     gnomad_AF oddsratio       pvalue  \\\n",
       "0  2.10420e-02   19.9388  7.84166e-09   \n",
       "1  1.37600e-02   26.0633  8.22806e-09   \n",
       "2  4.09360e-06   61070.8  1.40719e-23   \n",
       "3  3.05087e-03   50.2731  2.25792e-06   \n",
       "4  1.58466e-03    126.01   1.4352e-09   \n",
       "\n",
       "                                         annotations  \n",
       "0  CSQ=C|missense_variant|MODERATE|TRUB1|ENSG0000...  \n",
       "1  CSQ=-|intron_variant&non_coding_transcript_var...  \n",
       "2                                           segdup\\n  \n",
       "3                                           segdup\\n  \n",
       "4                                           segdup\\n  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = pd.DataFrame(optom_sig_gws).T.reset_index()\n",
    "test.rename(columns={'index': 'chr:bp', 0: 'rs_id', 1:'A1', 2:'A2', 3:'flag_optom', 4:'AC', 5:'AF', 6:'AN', \n",
    "                   7:'filter',  8:'case_AC', 9:'case_AN', 10:'control_AC', 11:'control_AN', \n",
    "                   12:'flag_gnomad', 13:'gnomad_AF', 14:'oddsratio', 15:'pvalue', 16:'annotations'}, inplace=True)\n",
    "test.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr:bp</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>flag_optom</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>filter</th>\n",
       "      <th>case_AC</th>\n",
       "      <th>case_AN</th>\n",
       "      <th>control_AC</th>\n",
       "      <th>control_AN</th>\n",
       "      <th>flag_gnomad</th>\n",
       "      <th>gnomad_AF</th>\n",
       "      <th>oddsratio</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>annotations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>785</th>\n",
       "      <td>16:33962400</td>\n",
       "      <td>rs74410326</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=30</td>\n",
       "      <td>AF=0.988</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>DB</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>69228</td>\n",
       "      <td>RF;AC0</td>\n",
       "      <td>0.00000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>1.63038e-113</td>\n",
       "      <td>segdup\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>784</th>\n",
       "      <td>16:33962391</td>\n",
       "      <td>rs77260646</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=30</td>\n",
       "      <td>AF=1.00</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>DB</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>6</td>\n",
       "      <td>70844</td>\n",
       "      <td>RF</td>\n",
       "      <td>8.46931e-05</td>\n",
       "      <td>inf</td>\n",
       "      <td>1.5895e-107</td>\n",
       "      <td>segdup\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1628</th>\n",
       "      <td>6:32486358</td>\n",
       "      <td>rs115773383</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=28</td>\n",
       "      <td>AF=0.800</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=3.33</td>\n",
       "      <td>28</td>\n",
       "      <td>30</td>\n",
       "      <td>1</td>\n",
       "      <td>68654</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1.45658e-05</td>\n",
       "      <td>961142</td>\n",
       "      <td>1.43038e-102</td>\n",
       "      <td>CSQ=C|synonymous_variant|LOW|HLA-DRB5|ENSG0000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1512</th>\n",
       "      <td>4:191003163</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>VQSRTrancheSNP99.00to99.90</td>\n",
       "      <td>AC=30</td>\n",
       "      <td>AF=1.00</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>DP=23</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>24938</td>\n",
       "      <td>RF;AC0</td>\n",
       "      <td>0.00000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>3.23383e-100</td>\n",
       "      <td>segdup\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1835</th>\n",
       "      <td>6:32610893</td>\n",
       "      <td>rs9272918</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=24</td>\n",
       "      <td>AF=0.782</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-1.208e+00</td>\n",
       "      <td>24</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>76970</td>\n",
       "      <td>RF;AC0</td>\n",
       "      <td>0.00000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>1.95933e-88</td>\n",
       "      <td>CSQ=A|3_prime_UTR_variant|MODIFIER|HLA-DQA1|EN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           chr:bp        rs_id A1 A2                   flag_optom     AC  \\\n",
       "785   16:33962400   rs74410326  C  A  VQSRTrancheSNP99.90to100.00  AC=30   \n",
       "784   16:33962391   rs77260646  C  A  VQSRTrancheSNP99.90to100.00  AC=30   \n",
       "1628   6:32486358  rs115773383  T  C  VQSRTrancheSNP99.90to100.00  AC=28   \n",
       "1512  4:191003163            .  G  T   VQSRTrancheSNP99.00to99.90  AC=30   \n",
       "1835   6:32610893    rs9272918  C  A  VQSRTrancheSNP99.90to100.00  AC=24   \n",
       "\n",
       "            AF     AN                   filter case_AC case_AN control_AC  \\\n",
       "785   AF=0.988  AN=30                       DB      30      30          0   \n",
       "784    AF=1.00  AN=30                       DB      30      30          6   \n",
       "1628  AF=0.800  AN=30        BaseQRankSum=3.33      28      30          1   \n",
       "1512   AF=1.00  AN=30                    DP=23      30      30          0   \n",
       "1835  AF=0.782  AN=30  BaseQRankSum=-1.208e+00      24      30          0   \n",
       "\n",
       "     control_AN flag_gnomad    gnomad_AF oddsratio        pvalue  \\\n",
       "785       69228      RF;AC0  0.00000e+00       inf  1.63038e-113   \n",
       "784       70844          RF  8.46931e-05       inf   1.5895e-107   \n",
       "1628      68654        PASS  1.45658e-05    961142  1.43038e-102   \n",
       "1512      24938      RF;AC0  0.00000e+00       inf  3.23383e-100   \n",
       "1835      76970      RF;AC0  0.00000e+00       inf   1.95933e-88   \n",
       "\n",
       "                                            annotations  \n",
       "785                                            segdup\\n  \n",
       "784                                            segdup\\n  \n",
       "1628  CSQ=C|synonymous_variant|LOW|HLA-DRB5|ENSG0000...  \n",
       "1512                                           segdup\\n  \n",
       "1835  CSQ=A|3_prime_UTR_variant|MODIFIER|HLA-DQA1|EN...  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test.sort_values(by=\"pvalue\").head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "test.to_csv('optom_gws_sig_results_exac.csv', index=False, header=True, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Grab optom genotype info for these variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "What do I want:\n",
    "    1) Number of Hets\n",
    "    2) Number of Homo for A2\n",
    "    3) Number of lowGQ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "with open(\"recalibrated_variants.postCGP.Gfiltered.Optom_subset.vcf\", 'rt') as f:\n",
    "    optom_geno = {}\n",
    "    info = []\n",
    "    # create a list of locations in optom\n",
    "    for line in f:\n",
    "        if line.startswith('#'):\n",
    "            continue\n",
    "        row = line.split('\\t')\n",
    "        tmp = [stuff.replace(\"chr\", \"\") for stuff in row]\n",
    "        ### create an index which is chr:bp\n",
    "        chr_bp = tmp[0:2]\n",
    "        index = ':'.join(map(str, chr_bp))\n",
    "        if index in optom_sig:\n",
    "            if index not in optom_geno:\n",
    "                homo_major = 0\n",
    "                hets = 0\n",
    "                homo_minor = 0\n",
    "                lowGQ_count = 0\n",
    "                for x in range(9,24) :\n",
    "                    data = tmp[x].split(':')\n",
    "                    if data[0] == \"0/0\" :\n",
    "                        homo_major += 1\n",
    "                    elif data[0] == \"0/1\" :\n",
    "                        hets += 1\n",
    "                    elif data[0] == \"1/1\":\n",
    "                        homo_minor += 1\n",
    "                    if data[3] == 'lowGQ':\n",
    "                        lowGQ_count += 1\n",
    "                info = [tmp[2], tmp[3], tmp[4], homo_major, hets, homo_minor, lowGQ_count]\n",
    "                optom_geno[index] = info "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr:bp</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>homo_major</th>\n",
       "      <th>hets</th>\n",
       "      <th>homo_minor</th>\n",
       "      <th>lowGQ_numbers</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:100011298</td>\n",
       "      <td>rs138062420</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:100144676</td>\n",
       "      <td>rs191156429</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10:100148117</td>\n",
       "      <td>rs147555747</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10:100150830</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10:100167323</td>\n",
       "      <td>rs79218397</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr:bp        rs_id A1 A2 homo_major hets homo_minor lowGQ_numbers\n",
       "0  10:100011298  rs138062420  G  A         15    0          0             0\n",
       "1  10:100144676  rs191156429  C  T         15    0          0             0\n",
       "2  10:100148117  rs147555747  C  T         14    1          0             0\n",
       "3  10:100150830            .  G  A         15    0          0             0\n",
       "4  10:100167323   rs79218397  G  A         14    1          0             0"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genos = pd.DataFrame(optom_geno).T.reset_index()\n",
    "genos.rename(columns={'index': 'chr:bp', 0: 'rs_id', 1:'A1', 2:'A2', 3:'homo_major', 4:'hets', \n",
    "                      5:'homo_minor', 6:'lowGQ_numbers' }, inplace=True)\n",
    "genos.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "genos.to_csv('optom_genotpyes_results_exac.csv', index=False, header=True, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Association with just EUR population in GNOMAD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Allele counts to keep\n",
    "list_names =[\"AC_NFE\", \"AN_NFE\", \"AF_NFE\"]\n",
    "pattern = re.compile(\"|\".join(list_names))\n",
    "\n",
    "#not_pass_qc_gnomad = {}\n",
    "\n",
    "with open(\"gnomad.exomes.r2.0.2.sites.vcf\", 'rt') as vcf_nfe:\n",
    "    gnomad_NFE = {}\n",
    "    for line in vcf_nfe:\n",
    "        if line.startswith('#'):\n",
    "            continue \n",
    "        row = line.split('\\t')\n",
    "        ### create an index which is chr:bp\n",
    "        chr_bp = row[0:2]\n",
    "        index = ':'.join(map(str, chr_bp))\n",
    "        ### keep only those that are in optom:\n",
    "        if index in optom:\n",
    "            #if row[6] != \"PASS\":\n",
    "            #    if index not in not_pass_qc_gnomad:\n",
    "            #        not_pass_qc_gnomad[index] = [ row[2], row[6] ]\n",
    "            #    continue\n",
    "            ### output index, rsnumber, a1, a2, info\n",
    "            output = [ row[2], row[3], row[4], row[6], row[-1].split(';')[-1] ]\n",
    "            info = row[7].split(';')\n",
    "            keep_info = [item for item in info if re.match(pattern, item)]\n",
    "            output.extend(keep_info)\n",
    "            #print(info)\n",
    "            if index not in gnomad_NFE:\n",
    "                gnomad_NFE[index] = output\n",
    "        elif row[2] in snp:\n",
    "            if index not in gnomad_NFE:\n",
    "                ### output index, rsnumber, a1, a2, info\n",
    "                output = [ index, row[2], row[3], row[4], row[6], row[-1].split(';')[-1] ]\n",
    "                info = row[7].split(';')\n",
    "                keep_info = [item for item in info if re.match(pattern, item)]\n",
    "                output.extend(keep_info)\n",
    "                gnomad_NFE[index] = output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "188436"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(gnomad_NFE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rs4041247',\n",
       " 'AAGAGAG',\n",
       " 'AAGAGAGAG,AAGAGAGAGAG,AAGAG,AAG,AAGAGAGAGAGAG,A',\n",
       " 'PASS',\n",
       " 'lcr\\n',\n",
       " 'AC_NFE=1021,346,1983,315,31,1033',\n",
       " 'AN_NFE=15044',\n",
       " 'AF_NFE=6.78676e-02,2.29992e-02,1.31813e-01,2.09386e-02,2.06062e-03,6.86652e-02']"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnomad_NFE['5:172745269']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "gnomad_Set = set(gnomad_NFE)\n",
    "optom_Set = set(optom)\n",
    "\n",
    "results_NFE = {}\n",
    "results_gws_NFE = {}\n",
    "bad_snps = {}\n",
    "for name in optom_Set.intersection(gnomad_Set):\n",
    "    if optom[name][1] == gnomad_NFE[name][1] and optom[name][2] == gnomad_NFE[name][2]:\n",
    "        if (re.split(\"=|,\", gnomad_NFE[name][7])[1] != \".\") and (float(re.split(\"=|,\", gnomad_NFE[name][7])[1]) <= 0.05):\n",
    "            #print(optom[name][1], gnomad[name][1], optom[name][2], gnomad[name][2])\n",
    "            case_AC = int(re.split(\"=|,\", optom[name][4])[1])\n",
    "            case_AN = int(optom[name][6].split('=')[1])\n",
    "            control_AC = int(re.split(\"=|,\", gnomad_NFE[name][5])[1])\n",
    "            control_AN = int(gnomad_NFE[name][6].split('=')[1])\n",
    "            #print(case_AC, case_AN, control_AC, control_AN)\n",
    "            oddsratio, pvalue = stats.fisher_exact([[case_AC, case_AN-case_AC], [control_AC, control_AN-control_AC]], alternative='two-sided')\n",
    "            info = [case_AC, case_AN, control_AC, control_AN, gnomad_NFE[name][3], re.split(\"=|,\", gnomad_NFE[name][7])[1], oddsratio, pvalue, gnomad_NFE[name][4]]\n",
    "            if name not in results_NFE:\n",
    "                results_NFE[name] = info\n",
    "            if pvalue < 0.000005 :\n",
    "                if name not in results_gws_NFE:\n",
    "                    results_gws_NFE[name] = info\n",
    "    else:\n",
    "        data = [optom[name][1],  optom[name][2], gnomad_NFE[name][1], gnomad_NFE[name][2]]\n",
    "        bad_snps[name] = data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "53939"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(bad_snps)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "bad_snps_still = {}\n",
    "\n",
    "for name in bad_snps:\n",
    "    if optom[name][1] == gnomad_NFE[name][1] and optom[name][2] == gnomad_NFE[name][2].split(',')[0] :\n",
    "        if (re.split(\"=|,\", gnomad_NFE[name][7])[1] != \".\") and (float(re.split(\"=|,\", gnomad_NFE[name][7])[1]) <= 0.05):\n",
    "            #print(optom[name][1], gnomad[name][1], optom[name][2], gnomad[name][2])\n",
    "            case_AC = int(re.split(\"=|,\", optom[name][4])[1])\n",
    "            case_AN = int(optom[name][6].split('=')[1])\n",
    "            control_AC = int(re.split(\"=|,\", gnomad_NFE[name][5])[1])\n",
    "            control_AN = int(gnomad_NFE[name][6].split('=')[1])\n",
    "            #print(case_AC, case_AN, control_AC, control_AN)\n",
    "            oddsratio, pvalue = stats.fisher_exact([[case_AC, case_AN-case_AC], [control_AC, control_AN-control_AC]], alternative='two-sided')\n",
    "            info = [case_AC, case_AN, control_AC, control_AN, gnomad_NFE[name][3], re.split(\"=|,\", gnomad_NFE[name][7])[1], oddsratio, pvalue, gnomad_NFE[name][4]]\n",
    "            if name not in results_NFE:\n",
    "                results_NFE[name] = info\n",
    "            if pvalue < 0.000005 :\n",
    "                if name not in results_gws_NFE:\n",
    "                    results_gws_NFE[name] = info\n",
    "    elif len(re.split(\"=|,\", optom[name][3])) > 2:\n",
    "        if optom[name][1] == gnomad_NFE[name][1] and optom[name][2].split(',')[0] == gnomad_NFE[name][2].split(',')[0] :\n",
    "            if (re.split(\"=|,\", gnomad_NFE[name][7])[1] != \".\") and (float(re.split(\"=|,\", gnomad_NFE[name][7])[1]) <= 0.05):\n",
    "                #print(optom[name][1], gnomad[name][1], optom[name][2], gnomad[name][2])\n",
    "                case_AC = int(re.split(\"=|,\", optom[name][4])[1])\n",
    "                case_AN = int(optom[name][6].split('=')[1])\n",
    "                control_AC = int(re.split(\"=|,\", gnomad_NFE[name][5])[1])\n",
    "                control_AN = int(gnomad_NFE[name][6].split('=')[1])\n",
    "                #print(case_AC, case_AN, control_AC, control_AN)\n",
    "                oddsratio, pvalue = stats.fisher_exact([[case_AC, case_AN-case_AC], [control_AC, control_AN-control_AC]], alternative='two-sided')\n",
    "                info = [case_AC, case_AN, control_AC, control_AN, gnomad_NFE[name][3], re.split(\"=|,\", gnomad_NFE[name][7])[1], oddsratio, pvalue, gnomad_NFE[name][4]]\n",
    "                new_name = name + \"_1\"\n",
    "                if new_name not in results_NFE:\n",
    "                    results_NFE[new_name] = info\n",
    "                if pvalue < 0.000005 :\n",
    "                    if new_name not in results_gws_NFE:\n",
    "                        results_gws_NFE[new_name] = info\n",
    "    elif len(re.split(\"=|,\", optom[name][3])) > 3:\n",
    "        if optom[name][1] == gnomad_NFE[name][1] and optom[name][2].split(',')[1] == gnomad_NFE[name][2].split(',')[1] :\n",
    "            if (re.split(\"=|,\", gnomad_NFE[name][7])[1] != \".\") and (float(re.split(\"=|,\", gnomad_NFE[name][7])[2]) <= 0.05):\n",
    "                #print(optom[name][1], gnomad[name][1], optom[name][2], gnomad[name][2])\n",
    "                case_AC = int(re.split(\"=|,\", optom[name][4])[2])\n",
    "                case_AN = int(optom[name][6].split('=')[1])\n",
    "                control_AC = int(re.split(\"=|,\", gnomad_NFE[name][5])[2])\n",
    "                control_AN = int(gnomad_NFE[name][6].split('=')[1])\n",
    "                #print(case_AC, case_AN, control_AC, control_AN)\n",
    "                oddsratio, pvalue = stats.fisher_exact([[case_AC, case_AN-case_AC], [control_AC, control_AN-control_AC]], alternative='two-sided')\n",
    "                info = [case_AC, case_AN, control_AC, control_AN, gnomad_NFE[name][3], re.split(\"=|,\", gnomad_NFE[name][7])[2], oddsratio, pvalue, gnomad_NFE[name][4]]\n",
    "                new_name = name + \"_2\"\n",
    "                if new_name not in results_NFE:\n",
    "                    results_NFE[new_name] = info\n",
    "                if pvalue < 0.000005 :\n",
    "                    if new_name not in results_gws_NFE:\n",
    "                        results_gws_NFE[new_name] = info   \n",
    "    elif len(re.split(\"=|,\", optom[name][3])) > 4:\n",
    "        if optom[name][1] == gnomad_NFE[name][1] and optom[name][2].split(',')[2] == gnomad_NFE[name][2].split(',')[2] :\n",
    "            if (re.split(\"=|,\", gnomad_NFE[name][7])[1] != \".\") and (float(re.split(\"=|,\", gnomad_NFE[name][7])[3]) <= 0.05):\n",
    "                #print(optom[name][1], gnomad[name][1], optom[name][2], gnomad[name][2])\n",
    "                case_AC = int(re.split(\"=|,\", optom[name][4])[3])\n",
    "                case_AN = int(optom[name][6].split('=')[1])\n",
    "                control_AC = int(re.split(\"=|,\", gnomad_NFE[name][5])[3])\n",
    "                control_AN = int(gnomad_NFE[name][6].split('=')[1])\n",
    "                #print(case_AC, case_AN, control_AC, control_AN)\n",
    "                oddsratio, pvalue = stats.fisher_exact([[case_AC, case_AN-case_AC], [control_AC, control_AN-control_AC]], alternative='two-sided')\n",
    "                info = [case_AC, case_AN, control_AC, control_AN, gnomad_NFE[name][3], re.split(\"=|,\", gnomad_NFE[name][7])[3], oddsratio, pvalue, gnomad_NFE[name][4]]\n",
    "                new_name = name + \"_3\"\n",
    "                if new_name not in results_NFE:\n",
    "                    results_NFE[new_name] = info\n",
    "                if pvalue < 0.000005 :\n",
    "                    if new_name not in results_gws_NFE:\n",
    "                        results_gws_NFE[new_name] = info  \n",
    "    else: \n",
    "        data = [optom[name][1],  optom[name][2], gnomad_NFE[name][1], gnomad_NFE[name][2]]\n",
    "        bad_snps_still[name] = data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "14070"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(bad_snps_still)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "all_keys = set(list(results_NFE.keys()) + list(optom.keys()))\n",
    "\n",
    "optom_sig_NFE = {}\n",
    "for key in all_keys:\n",
    "    if key in results_NFE and key in optom:\n",
    "        optom_sig_NFE[key] = optom[key] + results_NFE[key]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr:bp</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>flag_optom</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>filter</th>\n",
       "      <th>case_AC</th>\n",
       "      <th>case_AN</th>\n",
       "      <th>control_AC</th>\n",
       "      <th>control_AN</th>\n",
       "      <th>flag_gnomad</th>\n",
       "      <th>gnomad_AF</th>\n",
       "      <th>oddsratio</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>annotations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:100011298</td>\n",
       "      <td>rs138062420</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=0</td>\n",
       "      <td>AF=0.013</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-2.492e+00</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>234</td>\n",
       "      <td>110236</td>\n",
       "      <td>PASS</td>\n",
       "      <td>2.12272e-03</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>CSQ=A|intron_variant|MODIFIER|LOXL4|ENSG000001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:100144676</td>\n",
       "      <td>rs191156429</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=0</td>\n",
       "      <td>AF=0.013</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-6.391e+00</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>61</td>\n",
       "      <td>104180</td>\n",
       "      <td>PASS</td>\n",
       "      <td>5.85525e-04</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>CSQ=T|intron_variant|MODIFIER|PYROXD2|ENSG0000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10:100148117</td>\n",
       "      <td>rs147555747</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1</td>\n",
       "      <td>AF=0.013</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-2.965e+00</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>360</td>\n",
       "      <td>111504</td>\n",
       "      <td>PASS</td>\n",
       "      <td>3.22858e-03</td>\n",
       "      <td>10.646</td>\n",
       "      <td>0.0926895</td>\n",
       "      <td>CSQ=T|missense_variant|MODERATE|PYROXD2|ENSG00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10:100150830</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=0</td>\n",
       "      <td>AF=0.013</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-2.322e+00</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>121</td>\n",
       "      <td>110074</td>\n",
       "      <td>RF</td>\n",
       "      <td>1.09926e-03</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>CSQ=A|missense_variant|MODERATE|PYROXD2|ENSG00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10:100167323</td>\n",
       "      <td>rs79218397</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1</td>\n",
       "      <td>AF=0.013</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=1.96</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>27</td>\n",
       "      <td>110718</td>\n",
       "      <td>PASS</td>\n",
       "      <td>2.43863e-04</td>\n",
       "      <td>141.368</td>\n",
       "      <td>0.00755803</td>\n",
       "      <td>CSQ=A|intron_variant|MODIFIER|PYROXD2|ENSG0000...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr:bp        rs_id A1 A2 flag_optom    AC        AF     AN  \\\n",
       "0  10:100011298  rs138062420  G  A       PASS  AC=0  AF=0.013  AN=30   \n",
       "1  10:100144676  rs191156429  C  T       PASS  AC=0  AF=0.013  AN=30   \n",
       "2  10:100148117  rs147555747  C  T       PASS  AC=1  AF=0.013  AN=30   \n",
       "3  10:100150830            .  G  A       PASS  AC=0  AF=0.013  AN=30   \n",
       "4  10:100167323   rs79218397  G  A       PASS  AC=1  AF=0.013  AN=30   \n",
       "\n",
       "                    filter case_AC case_AN control_AC control_AN flag_gnomad  \\\n",
       "0  BaseQRankSum=-2.492e+00       0      30        234     110236        PASS   \n",
       "1  BaseQRankSum=-6.391e+00       0      30         61     104180        PASS   \n",
       "2  BaseQRankSum=-2.965e+00       1      30        360     111504        PASS   \n",
       "3  BaseQRankSum=-2.322e+00       0      30        121     110074          RF   \n",
       "4        BaseQRankSum=1.96       1      30         27     110718        PASS   \n",
       "\n",
       "     gnomad_AF oddsratio      pvalue  \\\n",
       "0  2.12272e-03         0           1   \n",
       "1  5.85525e-04         0           1   \n",
       "2  3.22858e-03    10.646   0.0926895   \n",
       "3  1.09926e-03         0           1   \n",
       "4  2.43863e-04   141.368  0.00755803   \n",
       "\n",
       "                                         annotations  \n",
       "0  CSQ=A|intron_variant|MODIFIER|LOXL4|ENSG000001...  \n",
       "1  CSQ=T|intron_variant|MODIFIER|PYROXD2|ENSG0000...  \n",
       "2  CSQ=T|missense_variant|MODERATE|PYROXD2|ENSG00...  \n",
       "3  CSQ=A|missense_variant|MODERATE|PYROXD2|ENSG00...  \n",
       "4  CSQ=A|intron_variant|MODIFIER|PYROXD2|ENSG0000...  "
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_NFE_sig = pd.DataFrame(optom_sig_NFE).T.reset_index()\n",
    "test_NFE_sig.rename(columns={'index': 'chr:bp', 0: 'rs_id', 1:'A1', 2:'A2', 3:'flag_optom', 4:'AC', 5:'AF', 6:'AN', \n",
    "                   7:'filter',  8:'case_AC', 9:'case_AN', 10:'control_AC', 11:'control_AN', \n",
    "                   12:'flag_gnomad', 13:'gnomad_AF', 14:'oddsratio', 15:'pvalue', 16:'annotations'}, inplace=True)\n",
    "test_NFE_sig.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "test_NFE_sig.to_csv('optom_results_exac_NFE.csv', index=False, header=True, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "all_keys = set(list(results_gws_NFE.keys()) + list(optom.keys()))\n",
    "\n",
    "optom_sig_gws_NFE = {}\n",
    "for key in all_keys:\n",
    "    if key in results_gws_NFE and key in optom:\n",
    "        optom_sig_gws_NFE[key] = optom[key] + results_gws_NFE[key]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2539"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(optom_sig_gws_NFE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr:bp</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>flag_optom</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>filter</th>\n",
       "      <th>case_AC</th>\n",
       "      <th>case_AN</th>\n",
       "      <th>control_AC</th>\n",
       "      <th>control_AN</th>\n",
       "      <th>flag_gnomad</th>\n",
       "      <th>gnomad_AF</th>\n",
       "      <th>oddsratio</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>annotations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:116698112</td>\n",
       "      <td>rs199739741</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=9</td>\n",
       "      <td>AF=0.113</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-3.965e+00</td>\n",
       "      <td>9</td>\n",
       "      <td>30</td>\n",
       "      <td>1572</td>\n",
       "      <td>69330</td>\n",
       "      <td>RF</td>\n",
       "      <td>2.26742e-02</td>\n",
       "      <td>18.4727</td>\n",
       "      <td>1.50896e-08</td>\n",
       "      <td>CSQ=C|missense_variant|MODERATE|TRUB1|ENSG0000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:118927198</td>\n",
       "      <td>.</td>\n",
       "      <td>TGGTCCCGGCGCCCGGCCTGTGTGGTGCTCAGATCAGACTTATCCA...</td>\n",
       "      <td>T</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=8</td>\n",
       "      <td>AF=0.100</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-1.360e-01</td>\n",
       "      <td>8</td>\n",
       "      <td>30</td>\n",
       "      <td>74</td>\n",
       "      <td>5516</td>\n",
       "      <td>RF</td>\n",
       "      <td>1.34155e-02</td>\n",
       "      <td>26.742</td>\n",
       "      <td>7.26155e-09</td>\n",
       "      <td>CSQ=-|intron_variant&amp;non_coding_transcript_var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10:135438959</td>\n",
       "      <td>.</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>VQSRTrancheINDEL99.90to100.00</td>\n",
       "      <td>AC=6</td>\n",
       "      <td>AF=0.225</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-3.050e-01</td>\n",
       "      <td>6</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>111166</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.00000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>2.26192e-22</td>\n",
       "      <td>segdup\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10:135439015</td>\n",
       "      <td>rs75470891</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=4</td>\n",
       "      <td>AF=0.075</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-6.700e-02</td>\n",
       "      <td>4</td>\n",
       "      <td>30</td>\n",
       "      <td>201</td>\n",
       "      <td>109326</td>\n",
       "      <td>RF</td>\n",
       "      <td>1.83854e-03</td>\n",
       "      <td>83.5247</td>\n",
       "      <td>3.16307e-07</td>\n",
       "      <td>segdup\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10:135439049</td>\n",
       "      <td>rs200347477</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=5</td>\n",
       "      <td>AF=0.113</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=1.89</td>\n",
       "      <td>5</td>\n",
       "      <td>30</td>\n",
       "      <td>115</td>\n",
       "      <td>104664</td>\n",
       "      <td>RF</td>\n",
       "      <td>1.09875e-03</td>\n",
       "      <td>181.824</td>\n",
       "      <td>2.53267e-10</td>\n",
       "      <td>segdup\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr:bp        rs_id  \\\n",
       "0  10:116698112  rs199739741   \n",
       "1  10:118927198            .   \n",
       "2  10:135438959            .   \n",
       "3  10:135439015   rs75470891   \n",
       "4  10:135439049  rs200347477   \n",
       "\n",
       "                                                  A1 A2  \\\n",
       "0                                                  A  C   \n",
       "1  TGGTCCCGGCGCCCGGCCTGTGTGGTGCTCAGATCAGACTTATCCA...  T   \n",
       "2                                                 AC  A   \n",
       "3                                                  T  C   \n",
       "4                                                  G  C   \n",
       "\n",
       "                      flag_optom    AC        AF     AN  \\\n",
       "0    VQSRTrancheSNP99.90to100.00  AC=9  AF=0.113  AN=30   \n",
       "1                           PASS  AC=8  AF=0.100  AN=30   \n",
       "2  VQSRTrancheINDEL99.90to100.00  AC=6  AF=0.225  AN=30   \n",
       "3    VQSRTrancheSNP99.90to100.00  AC=4  AF=0.075  AN=30   \n",
       "4    VQSRTrancheSNP99.90to100.00  AC=5  AF=0.113  AN=30   \n",
       "\n",
       "                    filter case_AC case_AN control_AC control_AN flag_gnomad  \\\n",
       "0  BaseQRankSum=-3.965e+00       9      30       1572      69330          RF   \n",
       "1  BaseQRankSum=-1.360e-01       8      30         74       5516          RF   \n",
       "2  BaseQRankSum=-3.050e-01       6      30          0     111166        PASS   \n",
       "3  BaseQRankSum=-6.700e-02       4      30        201     109326          RF   \n",
       "4        BaseQRankSum=1.89       5      30        115     104664          RF   \n",
       "\n",
       "     gnomad_AF oddsratio       pvalue  \\\n",
       "0  2.26742e-02   18.4727  1.50896e-08   \n",
       "1  1.34155e-02    26.742  7.26155e-09   \n",
       "2  0.00000e+00       inf  2.26192e-22   \n",
       "3  1.83854e-03   83.5247  3.16307e-07   \n",
       "4  1.09875e-03   181.824  2.53267e-10   \n",
       "\n",
       "                                         annotations  \n",
       "0  CSQ=C|missense_variant|MODERATE|TRUB1|ENSG0000...  \n",
       "1  CSQ=-|intron_variant&non_coding_transcript_var...  \n",
       "2                                           segdup\\n  \n",
       "3                                           segdup\\n  \n",
       "4                                           segdup\\n  "
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_NFE = pd.DataFrame(optom_sig_gws_NFE).T.reset_index()\n",
    "test_NFE.rename(columns={'index': 'chr:bp', 0: 'rs_id', 1:'A1', 2:'A2', 3:'flag_optom', 4:'AC', 5:'AF', 6:'AN', \n",
    "                   7:'filter',  8:'case_AC', 9:'case_AN', 10:'control_AC', 11:'control_AN', \n",
    "                   12:'flag_gnomad', 13:'gnomad_AF', 14:'oddsratio', 15:'pvalue', 16:'annotations'}, inplace=True)\n",
    "test_NFE.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr:bp</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>flag_optom</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>AN</th>\n",
       "      <th>filter</th>\n",
       "      <th>case_AC</th>\n",
       "      <th>case_AN</th>\n",
       "      <th>control_AC</th>\n",
       "      <th>control_AN</th>\n",
       "      <th>flag_gnomad</th>\n",
       "      <th>gnomad_AF</th>\n",
       "      <th>oddsratio</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>annotations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>787</th>\n",
       "      <td>16:33962400</td>\n",
       "      <td>rs74410326</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=30</td>\n",
       "      <td>AF=0.988</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>DB</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>23734</td>\n",
       "      <td>RF;AC0</td>\n",
       "      <td>0.00000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>1.4257e-99</td>\n",
       "      <td>segdup\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>786</th>\n",
       "      <td>16:33962391</td>\n",
       "      <td>rs77260646</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=30</td>\n",
       "      <td>AF=1.00</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>DB</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>23358</td>\n",
       "      <td>RF</td>\n",
       "      <td>0.00000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>2.30117e-99</td>\n",
       "      <td>segdup\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1611</th>\n",
       "      <td>6:32486358</td>\n",
       "      <td>rs115773383</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=28</td>\n",
       "      <td>AF=0.800</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=3.33</td>\n",
       "      <td>28</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>32194</td>\n",
       "      <td>PASS</td>\n",
       "      <td>0.00000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>7.91957e-95</td>\n",
       "      <td>CSQ=C|synonymous_variant|LOW|HLA-DRB5|ENSG0000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1494</th>\n",
       "      <td>4:191003163</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>VQSRTrancheSNP99.00to99.90</td>\n",
       "      <td>AC=30</td>\n",
       "      <td>AF=1.00</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>DP=23</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>8014</td>\n",
       "      <td>RF;AC0</td>\n",
       "      <td>0.00000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>1.91872e-85</td>\n",
       "      <td>segdup\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1822</th>\n",
       "      <td>6:32610893</td>\n",
       "      <td>rs9272918</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=24</td>\n",
       "      <td>AF=0.782</td>\n",
       "      <td>AN=30</td>\n",
       "      <td>BaseQRankSum=-1.208e+00</td>\n",
       "      <td>24</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>30596</td>\n",
       "      <td>RF;AC0</td>\n",
       "      <td>0.00000e+00</td>\n",
       "      <td>inf</td>\n",
       "      <td>8.01818e-79</td>\n",
       "      <td>CSQ=A|3_prime_UTR_variant|MODIFIER|HLA-DQA1|EN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           chr:bp        rs_id A1 A2                   flag_optom     AC  \\\n",
       "787   16:33962400   rs74410326  C  A  VQSRTrancheSNP99.90to100.00  AC=30   \n",
       "786   16:33962391   rs77260646  C  A  VQSRTrancheSNP99.90to100.00  AC=30   \n",
       "1611   6:32486358  rs115773383  T  C  VQSRTrancheSNP99.90to100.00  AC=28   \n",
       "1494  4:191003163            .  G  T   VQSRTrancheSNP99.00to99.90  AC=30   \n",
       "1822   6:32610893    rs9272918  C  A  VQSRTrancheSNP99.90to100.00  AC=24   \n",
       "\n",
       "            AF     AN                   filter case_AC case_AN control_AC  \\\n",
       "787   AF=0.988  AN=30                       DB      30      30          0   \n",
       "786    AF=1.00  AN=30                       DB      30      30          0   \n",
       "1611  AF=0.800  AN=30        BaseQRankSum=3.33      28      30          0   \n",
       "1494   AF=1.00  AN=30                    DP=23      30      30          0   \n",
       "1822  AF=0.782  AN=30  BaseQRankSum=-1.208e+00      24      30          0   \n",
       "\n",
       "     control_AN flag_gnomad    gnomad_AF oddsratio       pvalue  \\\n",
       "787       23734      RF;AC0  0.00000e+00       inf   1.4257e-99   \n",
       "786       23358          RF  0.00000e+00       inf  2.30117e-99   \n",
       "1611      32194        PASS  0.00000e+00       inf  7.91957e-95   \n",
       "1494       8014      RF;AC0  0.00000e+00       inf  1.91872e-85   \n",
       "1822      30596      RF;AC0  0.00000e+00       inf  8.01818e-79   \n",
       "\n",
       "                                            annotations  \n",
       "787                                            segdup\\n  \n",
       "786                                            segdup\\n  \n",
       "1611  CSQ=C|synonymous_variant|LOW|HLA-DRB5|ENSG0000...  \n",
       "1494                                           segdup\\n  \n",
       "1822  CSQ=A|3_prime_UTR_variant|MODIFIER|HLA-DQA1|EN...  "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_NFE.sort_values(by=\"pvalue\").head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "test_NFE.to_csv('optom_gws_sig_results_exac_NFE.csv', index=False, header=True, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Grab optom genotype info for these variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open(\"recalibrated_variants.postCGP.Gfiltered.Optom_subset.vcf\", 'rt') as f:\n",
    "    optom_geno_NFE = {}\n",
    "    info = []\n",
    "    # create a list of locations in optom\n",
    "    for line in f:\n",
    "        if line.startswith('#'):\n",
    "            continue\n",
    "        row = line.split('\\t')\n",
    "        tmp = [stuff.replace(\"chr\", \"\") for stuff in row]\n",
    "        ### create an index which is chr:bp\n",
    "        chr_bp = tmp[0:2]\n",
    "        index = ':'.join(map(str, chr_bp))\n",
    "        if index in optom_sig_NFE:\n",
    "            if index not in optom_geno_NFE:\n",
    "                homo_major = 0\n",
    "                hets = 0\n",
    "                homo_minor = 0\n",
    "                lowGQ_count = 0\n",
    "                for x in range(9,24) :\n",
    "                    data = tmp[x].split(':')\n",
    "                    if data[0] == \"0/0\" :\n",
    "                        homo_major += 1\n",
    "                    elif data[0] == \"0/1\" :\n",
    "                        hets += 1\n",
    "                    elif data[0] == \"1/1\":\n",
    "                        homo_minor += 1\n",
    "                    if data[3] == 'lowGQ':\n",
    "                        lowGQ_count += 1\n",
    "                info = [tmp[2], tmp[3], tmp[4], homo_major, hets, homo_minor, lowGQ_count]\n",
    "                optom_geno_NFE[index] = info "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr:bp</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>homo_major</th>\n",
       "      <th>hets</th>\n",
       "      <th>homo_minor</th>\n",
       "      <th>lowGQ_numbers</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:100011298</td>\n",
       "      <td>rs138062420</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:100144676</td>\n",
       "      <td>rs191156429</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10:100148117</td>\n",
       "      <td>rs147555747</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10:100150830</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10:100167323</td>\n",
       "      <td>rs79218397</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr:bp        rs_id A1 A2 homo_major hets homo_minor lowGQ_numbers\n",
       "0  10:100011298  rs138062420  G  A         15    0          0             0\n",
       "1  10:100144676  rs191156429  C  T         15    0          0             0\n",
       "2  10:100148117  rs147555747  C  T         14    1          0             0\n",
       "3  10:100150830            .  G  A         15    0          0             0\n",
       "4  10:100167323   rs79218397  G  A         14    1          0             0"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genos_NFE = pd.DataFrame(optom_geno_NFE).T.reset_index()\n",
    "genos_NFE.rename(columns={'index': 'chr:bp', 0: 'rs_id', 1:'A1', 2:'A2', 3:'homo_major', 4:'hets', \n",
    "                      5:'homo_minor', 6:'lowGQ_numbers'}, inplace=True)\n",
    "genos_NFE.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "genos_NFE.to_csv('optom_genotpyes_results_exac_NFE.csv', index=False, header=True, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [python3]",
   "language": "python",
   "name": "Python [python3]"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
